source/common/modules/markdown-editor/util/remove-markdown-link.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        removeMarkdownLink function
+ * CVM-Role:        Utility function
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This function removes a markdown link, performing necessary
+ *                  transformations where applicable.
+ *
+ * END HEADER
+ */
+
+import { type EditorView } from '@codemirror/view'
+import { type ChangeSpec } from '@codemirror/state'
+import { type SyntaxNode } from '@lezer/common'
+
+/**
+ * Utility to remove a markdown link. It extracts the text from the markdown
+ * link and replaces the link with just the text.
+ *
+ * @param {EditorView} view - The editor view
+ * @param {SyntaxNode} node - The node containing the Link
+ */
+export function removeMarkdownLink (node: SyntaxNode, view: EditorView): void {
+  // To deal with user being able to click on either part of the link (Text or URL)
+  // And the two different types of links []() and <>
+  // Check which node they clicked on before passing it
+  var useNode: SyntaxNode = node
+  if (node.type.name === 'URL' && node.parent != null) {
+    if (node.parent.type.name === 'Link') {
+      useNode = node.parent
+    } if (node.parent.type.name === 'Paragraph') {
```

Comment:
What exactly is the purpose of this specific check …? Did you verify what the syntax tree looks like for all types of links? The passed `node` should always be some type of link, correct …?

Reply:
No actually I found that there are 3 different cases of how a user can enter the right click context menu for a link:

For a [text](url) link:

1. They can right click on the [text] node which has type 'Link'
2. They can right click on the (url) node which has type 'Url'

For a <url> link:

3. They can right click on the <url> node which has type 'Url'

For cases 1. and 2. we extract the text from the 'Link' node, for case 3. we extract the text from the 'Url' node. I think the way which this was handled was a bit convoluted, I'll try simplifying this.

Reply:
Ah, got it, thanks!